src/System.IO.Packaging/src/System/IO/Packaging/ContentType.cs

```diff
@@ -231,8 +231,8 @@ internal bool AreTypeAndSubTypeEqual(ContentType contentType, bool allowParamete
                 // safe comparison because the _type and _subType strings have been restricted to
                 // ASCII characters, digits, and a small set of symbols.  This is not a safe comparison
                 // for the broader set of strings that have not been restricted in the same way.
-                result = (String.Compare(_type, contentType.TypeComponent, StringComparison.OrdinalIgnoreCase) == 0 &&
-                          String.Compare(_subType, contentType.SubTypeComponent, StringComparison.OrdinalIgnoreCase) == 0);
+                result = (_type.Equals(contentType.TypeComponent, StringComparison.OrdinalIgnoreCase) &&
+                          _subType.Equals(contentType.SubTypeComponent, StringComparison.OrdinalIgnoreCase));
```

Comment:
From code inspection it looks like neither ```_type``` nor ```_subType``` can ever be null, but I still worry a bit that we might be missing a case where they could be in, which case this could null ref.  I think it'd be better to use the static String.Equals method, e.g.
```
String.Equals(_type, contentType.TypeComponent, StringComparison.OrdinalIgnoreCase)
```
as this will similarly check for null as does Compare.